pkg/driver/driver_azure.go

```diff
@@ -1034,17 +1059,67 @@ func (clients *azureDriverClients) checkOrphanDisks(ctx context.Context, resourc
 	return nil
 }
 
+func (clients *azureDriverClients) checkNICStatus(ctx context.Context, resourceGroupName string, nicName string, shouldExist bool) func() error {
+	return func() error {
+		nic, err := clients.nic.Get(ctx, resourceGroupName, nicName, "")
+
+		// Case-1: If NIC should exist, check below if condition
+		if shouldExist {
+			if err == nil && nic.ID != nil {
+				// NIC exists
+				return nil
+			}
+
+			klog.V(4).Infof("NIC doesn't existance for %q", nicName)
```

Comment:
Can we keep the log as simple as NIC <nicName> not found? The current one does not seem to be so good. Just a thought.

Reply:
Done. Here is the change - https://github.com/gardener/machine-controller-manager/pull/594/commits/3c0e156a9f9200159cc852b45eb488cd6c153360.